File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/kafka_topic_schema.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.


from .schema import Schema
from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class KafkaTopicSchema(Schema):
    """
    Derived model for topic of a Kafka external catalog.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new KafkaTopicSchema object with values from keyword arguments. The default value of the :py:attr:`~oci.aidataplatform_dp.models.KafkaTopicSchema.entity_type` attribute
        of this class is ``KAFKA_TOPIC`` and it should not be changed.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param entity_type:
            The value to assign to the entity_type property of this KafkaTopicSchema.
            Allowed values for this property are: "ADW", "ALH", "STANDARD", "KAFKA_TOPIC", "ATP", "ORACLE", "EXADATA"
        :type entity_type: str

        :param key:
            The value to assign to the key property of this KafkaTopicSchema.
        :type key: str

        :param display_name:
            The value to assign to the display_name property of this KafkaTopicSchema.
        :type display_name: str

        :param description:
            The value to assign to the description property of this KafkaTopicSchema.
        :type description: str

        :param time_created:
            The value to assign to the time_created property of this KafkaTopicSchema.
        :type time_created: datetime

        :param time_updated:
            The value to assign to the time_updated property of this KafkaTopicSchema.
        :type time_updated: datetime

        :param created_by:
            The value to assign to the created_by property of this KafkaTopicSchema.
        :type created_by: str

        :param updated_by:
            The value to assign to the updated_by property of this KafkaTopicSchema.
        :type updated_by: str

        :param properties:
            The value to assign to the properties property of this KafkaTopicSchema.
        :type properties: dict(str, str)

        :param catalog_name:
            The value to assign to the catalog_name property of this KafkaTopicSchema.
        :type catalog_name: str

        :param lifecycle_state:
            The value to assign to the lifecycle_state property of this KafkaTopicSchema.
            Allowed values for this property are: "ACTIVE", "CREATING", "DELETING"
        :type lifecycle_state: str

        :param lifecycle_state_details:
            The value to assign to the lifecycle_state_details property of this KafkaTopicSchema.
        :type lifecycle_state_details: str

        :param details:
            The value to assign to the details property of this KafkaTopicSchema.
        :type details: dict(str, str)

        :param partitions:
            The value to assign to the partitions property of this KafkaTopicSchema.
        :type partitions: str

        :param replication_factor:
            The value to assign to the replication_factor property of this KafkaTopicSchema.
        :type replication_factor: str

        :param bootstrap_servers:
            The value to assign to the bootstrap_servers property of this KafkaTopicSchema.
        :type bootstrap_servers: str

        :param partition_details:
            The value to assign to the partition_details property of this KafkaTopicSchema.
        :type partition_details: str

        """
        self.swagger_types = {
            'entity_type': 'str',
            'key': 'str',
            'display_name': 'str',
            'description': 'str',
            'time_created': 'datetime',
            'time_updated': 'datetime',
            'created_by': 'str',
            'updated_by': 'str',
            'properties': 'dict(str, str)',
            'catalog_name': 'str',
            'lifecycle_state': 'str',
            'lifecycle_state_details': 'str',
            'details': 'dict(str, str)',
            'partitions': 'str',
            'replication_factor': 'str',
            'bootstrap_servers': 'str',
            'partition_details': 'str'
        }

        self.attribute_map = {
            'entity_type': 'entityType',
            'key': 'key',
            'display_name': 'displayName',
            'description': 'description',
            'time_created': 'timeCreated',
            'time_updated': 'timeUpdated',
            'created_by': 'createdBy',
            'updated_by': 'updatedBy',
            'properties': 'properties',
            'catalog_name': 'catalogName',
            'lifecycle_state': 'lifecycleState',
            'lifecycle_state_details': 'lifecycleStateDetails',
            'details': 'details',
            'partitions': 'partitions',
            'replication_factor': 'replicationFactor',
            'bootstrap_servers': 'bootstrapServers',
            'partition_details': 'partitionDetails'
        }

        self._entity_type = None
        self._key = None
        self._display_name = None
        self._description = None
        self._time_created = None
        self._time_updated = None
        self._created_by = None
        self._updated_by = None
        self._properties = None
        self._catalog_name = None
        self._lifecycle_state = None
        self._lifecycle_state_details = None
        self._details = None
        self._partitions = None
        self._replication_factor = None
        self._bootstrap_servers = None
        self._partition_details = None
        self._entity_type = 'KAFKA_TOPIC'

    @property
    def partitions(self):
        """
        Gets the partitions of this KafkaTopicSchema.
        The number of partitions in the Kafka topic.


        :return: The partitions of this KafkaTopicSchema.
        :rtype: str
        """
        return self._partitions

    @partitions.setter
    def partitions(self, partitions):
        """
        Sets the partitions of this KafkaTopicSchema.
        The number of partitions in the Kafka topic.


        :param partitions: The partitions of this KafkaTopicSchema.
        :type: str
        """
        self._partitions = partitions

    @property
    def replication_factor(self):
        """
        Gets the replication_factor of this KafkaTopicSchema.
        The replication factor.


        :return: The replication_factor of this KafkaTopicSchema.
        :rtype: str
        """
        return self._replication_factor

    @replication_factor.setter
    def replication_factor(self, replication_factor):
        """
        Sets the replication_factor of this KafkaTopicSchema.
        The replication factor.


        :param replication_factor: The replication_factor of this KafkaTopicSchema.
        :type: str
        """
        self._replication_factor = replication_factor

    @property
    def bootstrap_servers(self):
        """
        Gets the bootstrap_servers of this KafkaTopicSchema.
        Bootstrap servers for the Kafka topic.


        :return: The bootstrap_servers of this KafkaTopicSchema.
        :rtype: str
        """
        return self._bootstrap_servers

    @bootstrap_servers.setter
    def bootstrap_servers(self, bootstrap_servers):
        """
        Sets the bootstrap_servers of this KafkaTopicSchema.
        Bootstrap servers for the Kafka topic.


        :param bootstrap_servers: The bootstrap_servers of this KafkaTopicSchema.
        :type: str
        """
        self._bootstrap_servers = bootstrap_servers

    @property
    def partition_details(self):
        """
        **[Required]** Gets the partition_details of this KafkaTopicSchema.
        The details of the partitions in Kafka topic.


        :return: The partition_details of this KafkaTopicSchema.
        :rtype: str
        """
        return self._partition_details

    @partition_details.setter
    def partition_details(self, partition_details):
        """
        Sets the partition_details of this KafkaTopicSchema.
        The details of the partitions in Kafka topic.


        :param partition_details: The partition_details of this KafkaTopicSchema.
        :type: str
        """
        self._partition_details = partition_details

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/KafkaTopicSchema.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * Derived model for topic of a Kafka external catalog.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=KafkaTopicSchema.Builder.class)
@com.fasterxml.jackson.annotation.JsonTypeInfo(use=com.fasterxml.jackson.annotation.JsonTypeInfo.Id.NAME, include=com.fasterxml.jackson.annotation.JsonTypeInfo.As.PROPERTY, property="entityType")

public final class KafkaTopicSchema extends Schema {
    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
    @com.fasterxml.jackson.annotation.JsonProperty("key")
private String key;

public Builder key(String key) {
    this.key = key;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("displayName")
private String displayName;

public Builder displayName(String displayName) {
    this.displayName = displayName;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("description")
private String description;

public Builder description(String description) {
    this.description = description;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("timeCreated")
private java.util.Date timeCreated;

public Builder timeCreated(java.util.Date timeCreated) {
    this.timeCreated = timeCreated;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("timeUpdated")
private java.util.Date timeUpdated;

public Builder timeUpdated(java.util.Date timeUpdated) {
    this.timeUpdated = timeUpdated;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("createdBy")
private String createdBy;

public Builder createdBy(String createdBy) {
    this.createdBy = createdBy;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("updatedBy")
private String updatedBy;

public Builder updatedBy(String updatedBy) {
    this.updatedBy = updatedBy;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("properties")
private java.util.Map<String, String> properties;

public Builder properties(java.util.Map<String, String> properties) {
    this.properties = properties;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("catalogName")
private String catalogName;

public Builder catalogName(String catalogName) {
    this.catalogName = catalogName;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("lifecycleState")
private SchemaLifecycleState lifecycleState;

public Builder lifecycleState(SchemaLifecycleState lifecycleState) {
    this.lifecycleState = lifecycleState;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("lifecycleStateDetails")
private String lifecycleStateDetails;

public Builder lifecycleStateDetails(String lifecycleStateDetails) {
    this.lifecycleStateDetails = lifecycleStateDetails;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("details")
private java.util.Map<String, String> details;

public Builder details(java.util.Map<String, String> details) {
    this.details = details;
    return this;
}
            /**
     * The number of partitions in the Kafka topic.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("partitions")
private String partitions;

        /**
         * The number of partitions in the Kafka topic.
         * @param partitions the value to set
         * @return this builder
         **/
        

public Builder partitions(String partitions) {
    this.partitions = partitions;
    return this;
}
            /**
     * The replication factor.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("replicationFactor")
private String replicationFactor;

        /**
         * The replication factor.
         * @param replicationFactor the value to set
         * @return this builder
         **/
        

public Builder replicationFactor(String replicationFactor) {
    this.replicationFactor = replicationFactor;
    return this;
}
            /**
     * Bootstrap servers for the Kafka topic.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("bootstrapServers")
private String bootstrapServers;

        /**
         * Bootstrap servers for the Kafka topic.
         * @param bootstrapServers the value to set
         * @return this builder
         **/
        

public Builder bootstrapServers(String bootstrapServers) {
    this.bootstrapServers = bootstrapServers;
    return this;
}
            /**
     * The details of the partitions in Kafka topic.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("partitionDetails")
private String partitionDetails;

        /**
         * The details of the partitions in Kafka topic.
         * @param partitionDetails the value to set
         * @return this builder
         **/
        

public Builder partitionDetails(String partitionDetails) {
    this.partitionDetails = partitionDetails;
    return this;
}


        public KafkaTopicSchema build() {
            KafkaTopicSchema model = new KafkaTopicSchema(this.key
                , this.displayName
                , this.description
                , this.timeCreated
                , this.timeUpdated
                , this.createdBy
                , this.updatedBy
                , this.properties
                , this.catalogName
                , this.lifecycleState
                , this.lifecycleStateDetails
                , this.details
                , this.partitions
                , this.replicationFactor
                , this.bootstrapServers
                , this.partitionDetails);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(KafkaTopicSchema model) {
                this.key(model.getKey());
    this.displayName(model.getDisplayName());
    this.description(model.getDescription());
    this.timeCreated(model.getTimeCreated());
    this.timeUpdated(model.getTimeUpdated());
    this.createdBy(model.getCreatedBy());
    this.updatedBy(model.getUpdatedBy());
    this.properties(model.getProperties());
    this.catalogName(model.getCatalogName());
    this.lifecycleState(model.getLifecycleState());
    this.lifecycleStateDetails(model.getLifecycleStateDetails());
    this.details(model.getDetails());
    this.partitions(model.getPartitions());
    this.replicationFactor(model.getReplicationFactor());
    this.bootstrapServers(model.getBootstrapServers());
    this.partitionDetails(model.getPartitionDetails());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    
    @Deprecated
    public KafkaTopicSchema(String key, String displayName, String description, java.util.Date timeCreated, java.util.Date timeUpdated, String createdBy, String updatedBy, java.util.Map<String, String> properties, String catalogName, SchemaLifecycleState lifecycleState, String lifecycleStateDetails, java.util.Map<String, String> details, String partitions, String replicationFactor, String bootstrapServers, String partitionDetails) {
    super(key, displayName, description, timeCreated, timeUpdated, createdBy, updatedBy, properties, catalogName, lifecycleState, lifecycleStateDetails, details);
        this.partitions = partitions;
        this.replicationFactor = replicationFactor;
        this.bootstrapServers = bootstrapServers;
        this.partitionDetails = partitionDetails;
    }


        /**
     * The number of partitions in the Kafka topic.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("partitions")
    private final String partitions;

        /**
     * The number of partitions in the Kafka topic.
     * @return the value
     **/
    
    public String getPartitions() {
        return partitions;
    }


        /**
     * The replication factor.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("replicationFactor")
    private final String replicationFactor;

        /**
     * The replication factor.
     * @return the value
     **/
    
    public String getReplicationFactor() {
        return replicationFactor;
    }


        /**
     * Bootstrap servers for the Kafka topic.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("bootstrapServers")
    private final String bootstrapServers;

        /**
     * Bootstrap servers for the Kafka topic.
     * @return the value
     **/
    
    public String getBootstrapServers() {
        return bootstrapServers;
    }


        /**
     * The details of the partitions in Kafka topic.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("partitionDetails")
    private final String partitionDetails;

        /**
     * The details of the partitions in Kafka topic.
     * @return the value
     **/
    
    public String getPartitionDetails() {
        return partitionDetails;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("KafkaTopicSchema(");
        sb.append("super=").append(super.toString(includeByteArrayContents));
        sb.append(", partitions=").append(String.valueOf(this.partitions));
        sb.append(", replicationFactor=").append(String.valueOf(this.replicationFactor));
        sb.append(", bootstrapServers=").append(String.valueOf(this.bootstrapServers));
        sb.append(", partitionDetails=").append(String.valueOf(this.partitionDetails));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof KafkaTopicSchema)) {
            return false;
        }

        KafkaTopicSchema other = (KafkaTopicSchema) o;
        return java.util.Objects.equals(this.partitions, other.partitions) &&
            java.util.Objects.equals(this.replicationFactor, other.replicationFactor) &&
            java.util.Objects.equals(this.bootstrapServers, other.bootstrapServers) &&
            java.util.Objects.equals(this.partitionDetails, other.partitionDetails) &&
            super.equals(other);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = super.hashCode();
        result = (result * PRIME) + (this.partitions == null ? 43 : this.partitions.hashCode());
        result = (result * PRIME) + (this.replicationFactor == null ? 43 : this.replicationFactor.hashCode());
        result = (result * PRIME) + (this.bootstrapServers == null ? 43 : this.bootstrapServers.hashCode());
        result = (result * PRIME) + (this.partitionDetails == null ? 43 : this.partitionDetails.hashCode());
        return result;
    }


}
